DownloadManager/Plugin.DownloadManager.Android/DownloadFileImplementation.cs

```diff
@@ -92,20 +94,44 @@ public DownloadFileImplementation (ICursor cursor)
             Url = cursor.GetString (cursor.GetColumnIndex (Android.App.DownloadManager.ColumnUri));
         }
 
-        public void StartDownload (Android.App.DownloadManager downloadManager, string destinationPathName)
+        private bool _mobileNetworkAllowed = true;
+
+        public bool MobileNetworkAllowed
         {
-            using (var downloadUrl = Uri.Parse (Url))
-            using (var request = new Android.App.DownloadManager.Request (downloadUrl)) {
+            get
+            {
+                return _mobileNetworkAllowed;
+            }
+            set
+            {
+                _mobileNetworkAllowed = value;
+                if (Request != null && value == false) {
+                    Request.SetAllowedNetworkTypes(Android.App.DownloadNetwork.Wifi);
+                }
+            }
+        }
+
+        public void StartDownload (Android.App.DownloadManager downloadManager, string destinationPathName, bool? mobileNetworkAllowed)
+        {
+            using (var downloadUrl = Uri.Parse(Url))
+            using (Request = new Android.App.DownloadManager.Request(downloadUrl))
+            {
+
+                if (mobileNetworkAllowed != null) {
+                    MobileNetworkAllowed = (bool)mobileNetworkAllowed;
+                } else {
+                    MobileNetworkAllowed = MobileNetworkAllowed;
```

Comment:
Looks strange ... and just works because of some trigger in the setter of this variable ...